crates/context/primitives/src/client.rs

```diff
@@ -738,9 +742,61 @@ impl ContextClient {
             &types::ContextDelta::Data(delta.into()),
         )?;
 
+        // Prune old deltas to prevent unbounded storage growth
+        self.prune_state_delta(context_id, public_key, height);
+
         Ok(())
     }
 
+    /// Prunes old state deltas to prevent unbounded storage growth.
+    ///
+    /// Keeps only the last 256 deltas per member. Older deltas are automatically deleted.
+    /// Deletion errors are logged but do not cause the operation to fail, as the delta
+    /// may have already been pruned or deleted.
+    ///
+    /// # Arguments
+    ///
+    /// * `context_id` - The context ID for the delta.
+    /// * `public_key` - The member whose deltas are being pruned.
+    /// * `current_height` - The current height to prune from.
+    fn prune_state_delta(
+        &self,
+        context_id: &ContextId,
+        public_key: &PublicKey,
+        current_height: &NonZeroUsize,
+    ) {
+        const MAX_DELTAS_PER_MEMBER: usize = 256;
```

Comment:
it's better to pass the `max_deltas_per_member` as a parameter, and then you can pass that specific constant from `handle_state_delta`. This way it's easier for testing - e.g. for tests you would be able to issue 3 deltas and set the limit to 3 (instead of 256).